src/app/forms/services/dynamic-form.types.ts

```diff
@@ -192,11 +210,32 @@ export class CustomFormArray extends FormArray implements CustomArray, BaseForm
     return this.valueChanges.pipe(map(() => this.getCleanValue(this)));
   }
 
-  addControl(data?: any): void {
+  addControl(data?: any, templateArray?: FormNode): void {
+    if (templateArray) {
+      super.push(this.dynamicFormService.createForm(templateArray, data));
+      return;
+    }
     if (this.meta?.children) {
       super.push(this.dynamicFormService.createForm(this.meta.children[0], data));
     }
   }
+
+  override patchValue(
+    value: any[],
+    options?: {
+      onlySelf?: boolean;
+      emitEvent?: boolean;
+    }
+  ): void {
+    if (value.length !== this.controls.length) {
```

Comment:
How does this behave when patching on form groups? Maybe check controls is an array?
Does this only apply to the top level controls? What if there are nested arrays?

Reply:
This will never be called on a group as it's extending formArray, the group gets patched by a different function and does not need changing (it's that big comment block above!). So controls can only ever be an array. I guess that yes this will only manage non nested arrays, but do we need to extend the functionality to manage nested arrays? Not sure we can even make nested form arrays?